### ISSUE TITLE
"Connection refused" with Java 1.8

### ISSUE THREAD

**Issue by eigenhombre** [pjstadig/lein-maven-s3-wagon#4]
Hi!  I've been getting failures to connect to my S3 repo since I upgraded Java to 1.8:

```
(defproject s3bug "0.1.0-SNAPSHOT"
  :dependencies [[myorganization/mylibrary "0.4.19"]
                 [org.clojure/clojure "1.7.0"]]
  :repositories [["heimat" {:url "s3://mybucket/mydirectory/
                            :username :env/my_aws_keyid
                            :password :env/my_aws_secret}]]
  :plugins [[lein-maven-s3-wagon "0.2.4"]])

s3bug (master)  lein deps
[WARN] s3://mybucket/mydirectory/ - Connection refused
[INFO] Logged off - mybucket
Could not transfer artifact myorganization:mylibrary:pom:0.4.19 from/to heimat (s3://mybucket/mydirectory/): Could not connect to repository
This could be due to a typo in :dependencies or network issues.
If you are behind a proxy, try setting the 'http_proxy' environment variable.
```

If I switch back to Java 1.7, it works fine.

**Comment by wgb** [comment]
Also getting this same issue. I believe it's the latest Java 1.8 release as I was running 8 previously with no problem.

**Comment by blischalk** [comment]
I am encountering this issue as well.

**Comment by savagematt** [comment]
+1

**Comment by cch1** [comment]
Same here.

**Comment by pjstadig** [comment]
I believe this is fixed by #6 and has been released as 0.2.5. Can you try using that and verify that it fixes the issue?

**Comment by cch1** [comment]
I can confirm that the issue is resolved with release `0.2.5`.  Thanks @pjstadig
